laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -249,109 +249,57 @@ public function getDay($day = null, $room = null, $leadSurgeon = null, $rotation
-        $prevThird = Option::where('date', $date)->where('resident', $resident)->where('option', 3)->where('isValid', 1)->get();
-        if (sizeof($prevFirst) > 0){
-            $previous[0] = array(
-                'prevPref'=> $prevFirst,
-                'schedule'=> ScheduleData::where('id', $prevFirst[0]['schedule'])->get(),
-                'milestone'=> Milestone::where('id', $prevFirst[0]['milestones'])->get(),
-                'pref_anest'=> Anesthesiologist::where('id', $prevFirst[0]['anesthesiologist_id'])->get()
-            );
-        } else {
-            $previous[0] = NULL;
-        }
-
-        if (sizeof($prevSecond) > 0){
-            $previous[1] = array(
-                'prevPref'=> $prevSecond,
-                'schedule'=> ScheduleData::where('id', $prevSecond[0]['schedule'])->get(),
-                'milestone'=> Milestone::where('id', $prevSecond[0]['milestones'])->get(),
-                'pref_anest'=> Anesthesiologist::where('id', $prevSecond[0]['anesthesiologist_id'])->get()
-            );
-        } else {
-            $previous[1] = NULL;
-        }
-
-        if (sizeof($prevThird) > 0){
-            $previous[2] = array(
-                'prevPref'=> $prevThird,
-                'schedule'=> ScheduleData::where('id', $prevThird[0]['schedule'])->get(),
-                'milestone'=> Milestone::where('id', $prevThird[0]['milestones'])->get(),
-                'pref_anest'=> Anesthesiologist::where('id', $prevThird[0]['anesthesiologist_id'])->get()
-            );
-        } else {
-            $previous[2] = NULL;
+        $date = $currentChoices[0]['schedule'][0]['date'];
+        $previousChoices = array();
+        $resident = Resident::where('email', $_SERVER["HTTP_EMAIL"])->value('id');
+        for($i = 0; $i < 3; $i++) {
+            $previousOption = Option::where('date', $date)->where('resident', $resident)->where('option', ($i+1))->where('isValid', 1)->get();
+            if(sizeof($previousOption) > 0) {
+                $schedule = ScheduleData::where('id', $previousOption[0]['schedule'])->get();
+                $previousChoices[$i] = array(
+                    'schedule' => $schedule,
+                    'case_procedure' => self::parseCaseProcedure($schedule[0]['case_procedure']),
+                    'milestone' => Milestone::where('id', $previousOption[0]['milestones'])->get(),
+                    'objective' => $previousOption[0]['objectives'],
+                    'anesthesiologist_pref'=> Anesthesiologist::where('id', $previousOption[0]['anesthesiologist_id'])->get()
+                );
+            } else {
+                $previousChoices[$i] = NULL;
+            }
         }
+        return view('schedules.resident.schedule_confirm', compact('id', 'currentChoices', 'previousChoices', 'choiceTypes'));
+    }
 
-        return view('schedules.resident.schedule_confirm', compact('id', 'input', 'previous'));
+    public function parseCaseProcedure($case){
```

Comment:
This method might be extracted into a general file for helper methods. Same logic appears in `processSingleChoice($id)` in PagesController.php

Reply:
in the interest of keeping PRs small and focused, I'd prefer to see that refactoring separately, but if you think it's significantly easier to implement here, I trust your judgement.

Reply:
This logic was previously php in the blade

Reply:
I agree that extracting this method to general file should be a separate task depending on what we want to do about general methods like schedule data logic

Reply:
Makes sense. We might want to do something like creating a [helpers file](https://laravel-news.com/creating-helpers).